File: apps/ruby-sdk/lib/firecrawl/models/concurrency_check.rb
# frozen_string_literal: true

module Firecrawl
  module Models
    # Current concurrency usage.
    class ConcurrencyCheck
      attr_reader :concurrency, :max_concurrency

      def initialize(data)
        @concurrency = data["concurrency"]
        @max_concurrency = data["maxConcurrency"]
      end

      def to_s
        "ConcurrencyCheck{concurrency=#{concurrency}/#{max_concurrency}}"
      end
    end
  end
end

File: apps/dot-net-sdk/Firecrawl/Models/ConcurrencyCheck.cs
using System.Text.Json.Serialization;

namespace Firecrawl.Models;

/// <summary>
/// Current concurrency usage information.
/// </summary>
public class ConcurrencyCheck
{
    [JsonPropertyName("current")]
    public int Current { get; set; }

    [JsonPropertyName("max")]
    public int MaxConcurrency { get; set; }
}
